services/app-web/src/pages/Wrapper/AuthenticatedRouteWrapper.tsx

```diff
@@ -0,0 +1,113 @@
+import { dayjs } from '../../common-code/dateHelpers/dayjs'
+import { useLDClient } from 'launchdarkly-react-client-sdk'
+import { featureFlags } from '../../common-code/featureFlags'
+
+export const AuthenticatedRouteWrapper = ({
+    children,
+    setAlert,
+    authMode,
+}: {
+    children: React.ReactNode
+    setAlert?: React.Dispatch<React.ReactElement>
+    authMode: AuthModeType
+}): React.ReactElement => {
+    const { logout, isSessionExpiring, timeUntilLogout, updateSessionExpiry } =
+        useAuth()
+    const ldClient = useLDClient()
+    const minutesUntilExpiration = ldClient?.variation(
+        featureFlags.MINUTES_UNTIL_SESSION_EXPIRES,
+        30
+    )
+    const modalRef = createRef<ModalRef>()
+    const LocalStorage = window.localStorage
+    const logoutSession = useCallback(() => {
+        updateSessionExpiry(false)
+        LocalStorage.removeItem('LOGOUT_TIMER')
+        if (logout) {
+            logout().catch((e) => {
+                console.log('Error with logout: ', e)
+                setAlert &&
+                    setAlert(
+                        <Alert
+                            data-testid="Error400"
+                            style={{ width: '600px', marginBottom: '5px' }}
+                            type="error"
+                            heading="Oops! Something went wrong"
+                        />
+                    )
+            })
+        }
+    }, [logout, setAlert, updateSessionExpiry, LocalStorage])
+
+    const resetSessionTimeout = () => {
+        updateSessionExpiry(false)
+        const sessionExpirationTime = dayjs(Date.now()).add(
+            minutesUntilExpiration,
+            'minute'
+        )
+        try {
+            LocalStorage.setItem(
+                'LOGOUT_TIMER',
+                sessionExpirationTime.toISOString()
+            )
+        } catch (e) {
+            console.log('Error setting logout timer: ', e)
+        }
+        if (authMode !== 'LOCAL') {
+            void extendSession()
+        }
+    }
+
+    useEffect(() => {
+        modalRef.current?.toggleModal(undefined, isSessionExpiring)
+    }, [isSessionExpiring, modalRef])
+
+    useEffect(() => {
+        if (timeUntilLogout < 1) {
+            logoutSession()
+        }
+    }, [timeUntilLogout, logoutSession])
+    return (
+        <div className={styles.wrapper}>
+            {' '}
+            {
+                <Modal
+                    modalRef={modalRef}
+                    id="extend-session-modal"
+                    modalHeading="Session Expiring"
+                    onSubmitText="Continue Session"
+                    onCancelText="Logout"
+                    onCancel={logoutSession}
+                    submitButtonProps={{ className: styles.submitButton }}
+                    onSubmit={resetSessionTimeout}
+                    className={styles.expireModal}
+                    forceAction={true}
+                >
+                    <p>
```

Comment:
I think we need to you some kind of `aria-live` here - I'm going off Chris's OTT from the easi project where he showed (after user testing sessions) that with aria-live users with screenreaders get properly notified about whats going on on with the countdown. 

Here's the code that adds this in their app https://github.com/CMSgov/easi-app/pull/304/files#diff-ec7778003924faeebbb2bd218f1005e66c8d085c44365afde23614aef3e7a693R124-R128

Reply:
Also regarding screenreader testing - when I tested this I saw something interesting where when the modal first opens it also re-announces the page heading.  It might be because the React context is remounting he entire tree here and that is triggering `usePage` again? 🔍 I can poke around more if its helpful

Reply:
Good call.  👀

Reply:
I pushed a fix; let me know how it works for you.  And I'm not hearing the page heading thing.  We can chat Monday...

Reply:
🌮 I am hearing the announcement now 🌮 . 

nit (not blocking): The  initial announcement is a bit  - right now it says "Your session is going to expire two o'clock"  which sounds like a time.  For everything else it says one fourty, one thirty, etc so it sounds fine. Its only the 2:00 and 1:00
that have this issue.

I think this code might be causing it 
```
 Your session is going to expire in{' '}
                        {dayjs
                            .duration(timeUntilLogout, 'seconds')
                            .format('mm:ss')}{' '}
```
 fwiw I'm not seeing the double announce of the page heading when the modal opens on JAWS today (i had used voiceover on Friday).  I don't think its an issue